extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -86,7 +87,20 @@ require([
 	// fetch data for all recirculation modules
 	// TODO lazy load some data on scroll
 	liftigniter.fetch('ns');
-	liftigniter.fetch('wiki');
+
+	window.addEventListener('scroll', function () {
+		var mcfOffset = $mixedContentFooter.offset().top,
+			scrollPosition = $(window).scrollTop(),
+			windowInnerHeight = $(window).innerHeight(),
+			lazyLoadOffset = 500,
+			aproachingMCF = scrollPosition + windowInnerHeight + lazyLoadOffset > mcfOffset;
+
+		if (aproachingMCF && !wikiArticlesFetched) {
+			liftigniter.fetch('wiki');
+			wikiArticlesFetched = true;
+		}
+	});
+
 	discussions.fetch();
 
 	// TODO handle errors
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)